### ISSUE TITLE
A JavaScript error occurred in the main process

### ISSUE THREAD

**Issue by jacklenox** [Zettlr/Zettlr#34]
## Description

Similar to #14, I'm getting these errors across each directory, and each file in a directory upon opening Zettlr, or when adding any new directories. The error continues:

```
Uncaught Exception:
Error: watch /home/jacklenox/Documents/Zettlr ENOSPC
    at _errnoException (util.js:1024:11)
    at FSWatcher.start (fs.js:1386:19)
    at Object.fs.watch (fs.js:1412:11)
...
```

(Let me know if you'd like full output. I can't copy/paste for some reason.)

## Reproducing

It happens when opening Zettlr with any directory opened, or when adding new directories to Zettlr.

## Expected behaviour

No errors.

**Screenshots**

![screenshot from 2018-09-14 19-36-10](https://user-images.githubusercontent.com/751476/45565921-7d606380-b855-11e8-951d-b1b28e4baf3f.png)

![screenshot from 2018-09-14 19-38-31](https://user-images.githubusercontent.com/751476/45566031-c9aba380-b855-11e8-8d41-157da20fdc6c.png)

## Platform
 - OS: Pop!_OS
 - OS Version: 18.04 LTS (based on Ubuntu)
 - Zettlr Version: 0.19

**Comment by nathanlesage** [comment]
Hey,

I start to really dislike watching files using chokidar …

So the most obvious follow-up questions would be: Are there *a lot* (and I mean: A LOT) files in this directory? I suppose not, but just for clarification.

And then: Are there some strange chmod-settings on any of the files? (If your user account does not have read-permissions on a file, it may fail in this way, although this more likely results in an EPERM-error …)

Or any other idea that might point towards it?

Cheers!

**Comment by jacklenox** [comment]
Hey Nathan, thank you for the quick response!

Haha, I haven't dug into it too much myself yet.

> Are there *a lot* (and I mean: A LOT) files in this directory? I suppose
not, but just for clarification.

I'm afraid not. Just the one. And it happens whichever directory I open. I
even created a new directory in the app with one file in it, and when I
opened it this error still fired.

> Are there some strange chmod-settings on any of the files?

I don't think so. Just in my regular user directory, with standard
permissions.

I'll have a deeper look later. I should have said that after dismissing
each error by pressing "Ok", the app works fine. Great work, I love it!

Cheers.

**Comment by nathanlesage** [comment]
> I'm afraid not. Just the one. And it happens whichever directory I open. I even created a new directory in the app with one file in it, and when I opened it this error still fired.

So it indeed seems that it's some hiccup inside chokidar. Well, the package has had its problems. I'll look into possible alternatives once I finish some other work! But I may have an idea to catch the error to alleviate your pain (and that of others experiencing the same problem) until I find a viable solution!

> I should have said that after dismissing each error by pressing "Ok", the app works fine.

Understandably — the only thing is that it then doesn't watch your directory for remote changes.

I'll get back to you once I manage to find something!

P.S.: To all others, please share your ENOSPC errors, in case you encounter them. Just for me to get a picture of how many linux OS's are actually affected.

**Comment by escapedsalmon** [comment]
So, there we go:
Today I set up Ubuntu 18.04 LTS and the current Pop_OS! 18.10 release. Then I testet both the 0.20.0 .debs as well als the current develop branch on both systems.
I experienced some other strange things (which are off topic here) - but I was not able to reproduce or verify this issue at this time.

**Comment by jacklenox** [comment]
@escapedsalmon Apologies for the slow response. I actually didn't notice that this seemed to have fixed itself, but I have been happily using Zettlr without this error for the past month or so. It looks as though updating from Pop_OS 18.04 to 18.10 must have fixed this for me. I think that's the only notable thing I've done since I first reported this issue! Thank you for digging in.

**Comment by escapedsalmon** [comment]
@jacklenox This somehow fits into the pattern I expected. I assumed it might have to do with some security settings in AppArmor or some other security enhancements Ubuntu is using - but it was just a guess (knowing that SELinux is a common source of unknown and completely unrelated errors on Fedora).
Even though there's still a small chance @nathanlesage fixed the Bug in 0.20. without noticing it.

**Comment by nathanlesage** [comment]
@jacklenox Thanks a lot for the response!

As I did not touch the watchdog for months, I assume that @escapedsalmon is correct in stating that something in the security of your OS was not working correctly, which in turn prompted these errors.

But it's good to know it works now! :)
